src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -251,22 +250,28 @@ protected final void verify(Checker checker,
             for (int i = 0; i < expected.length; i++) {
                 final String expectedResult = messageFileName + ":" + expected[i];
                 final String actual = lnr.readLine();
-                assertEquals(expectedResult, actual, "error message " + i);
+                assertWithMessage("error message " + i)
```

Comment:
It is better to use formatted message here: 
```java
assertWithMessage("error message %s", i)
```

Reply:
What is better about it? The readability or is there some performance?

Reply:
Both `assertWithMessage("error message %s", i)` and `assertWithMessage("error message " + i) have the same readability and performance.

The difference is that `"error message " + i` is actually `String.concat("error message "), i.toString())`. And this `.toString()' method can (in theory) have side effects. For example, it may throw an NPE, which will cause the test to fail for no clear reason.

This is absolutely inapplicable to any primitive type such as `int`, but this is a style issue. If we all agree to use this style, it will avoid possible problems in the future.

Reply:
What about https://github.com/checkstyle/checkstyle/pull/11151#discussion_r777310160 ?